### ISSUE TITLE
Provide a Purl in the root component of the SBOM

### ISSUE THREAD

**Issue by MeikelVielhauer** [anchore/syft#1408]
**What would you like to be added**:
A Package Url from the docker image is missing in the generated SBOM.
Could you please provide the purl for the root component (metadata.component.purl)?

Here you can find examples for purls from docker images: https://github.com/package-url/purl-spec#some-purl-examples.

**Why is this needed**:
This is needed to have a specific identifier for an artifact deployed from a docker image.

**Additional conext**:
There is an "Executive Order on Improving the Nation’s Cybersecurity" providing urgency about delivering sboms for your published products to be compliant. For docker images, you should be able to map your archived sboms to your deployed artifacts via an identifier (purl).

**Comment by wagoodman** [comment]
Great idea! 

In the internal model this could probably go in the `source.Metadata` (needs more thought), and for each format we'd need to identify the correct location in the SBOM (seems like you're referring to CycloneDX for `metadata.component.purl`, yes?).

**Comment by MeikelVielhauer** [comment]
Yes, you're right, I am referring to the CycloneDx format.

**Comment by MeikelVielhauer** [comment]
Hello @wagoodman,
is there anything planned by your side related to this?

**Comment by MeikelVielhauer** [comment]
> Hello @wagoodman, is there anything planned by your side related to this?

I have not seen that you added this to your backlog. Sorry for the inconvenience, thanks a lot! 🚀

**Comment by kzantow** [comment]
Yes, we've added this to the backlog and will get to it in due time. Thanks, @MeikelVielhauer!

**Comment by Yaytay** [comment]
I'd very much like to see the logic from BuildKit's [PURLtoRef](https://github.com/moby/buildkit/blob/824bc35b9771357e548ca913f8790854b70cf1f5/util/purl/image.go#L60) added to syft.

**Comment by anilkeshav27** [comment]
@kzantow , @wagoodman , wanted to revist this topic, since we still docker sbom without a `metadata.component.purl `in the cycloneDx format . is this planned ?

**Comment by spiffcs** [comment]
I added needs discussion to this so we could talk about adding the RefToPurl from build kit into syft and what it might look like on the next livestream

**Comment by kzantow** [comment]
We _are_ [outputting OCI PURLs in SPDX](https://github.com/anchore/syft/blob/ca21ccf21d19ac183917f28f26457c674ff663b3/syft/format/common/spdxhelpers/to_format_model.go#L210) -- we should do the same thing in CycloneDX 👍
